mindmeld/ser.py

```diff
@@ -126,28 +131,65 @@ def parse_numerics(sentence, dimensions=None, language='EN', locale='en_US',
             * response_code (int): http status code.
     """
     if sentence == '':
+        logger.error('Empty query passed to the system entity resolver')
         return {}, SUCCESSFUL_HTTP_CODE
 
     data = {
         'text': sentence,
-        'lang': language,
         'latent': True,
     }
 
-    valid_locales = ["en_AU", "en_BE", "en_BZ", "en_CA", "en_CN", "en_GB", "en_HK", "en_IE",
-                     "en_IN", "en_JM", "en_MO", "en_NZ", "en_PH", "en_TT", "en_TW", "en_US",
-                     "en_ZA"]
+    if not language:
+        language = 'EN'
+
+    # uppercase the language code if it is not
+    language = language.upper()
+
+    if len(language) < 2 or len(language) > 3:
+        logger.error('System entity resolver only supports ISO 639-1 language codes')
+        language = None
+    elif len(language) == 3:
+        # If a ISO 639-2 code is provided, we attempt to convert it to
+        # ISO 639-1 since the dependent system entity resolver requires this
+        iso639_2_code = pycountry.languages.get(alpha_3=language.lower())
+        try:
+            language = getattr(iso639_2_code, 'alpha_2').upper()
+        except AttributeError:
+            language = None
+
+    # Set the default locale to be en_us if the locale is not passed in and the langauge code
+    # is english
+    if not locale and language == 'EN':
+        locale = 'en_US'
+
+    if locale:
+        language_code_of_locale = locale.split('_')[0]
+        if language != language_code_of_locale.upper():
+            logger.error('Language code %s and Locale code do not match %s, '
+                         'using only the locale code for processing', language, locale)
+            # The system entity recognizer prefers the locale code over the language code,
+            # so we bias towards sending just the locale code when the codes dont match.
+            language = None
+
+    if not locale and not language:
+        logger.error('System entity resolver only supports ISO 639-1 language codes '
+                     'and locales representing the ISO 639-1 language code and '
+                     'ISO3166 alpha 2 country code separated by an underscore '
+                     'character')
+        return {}, SUCCESSFUL_HTTP_CODE
```

Comment:
This seems similar but not quite the same as `_validate_language_code()` in `mindmeld.components.request`.

Is there any way we can share some of that code so we can test it once?

Reply:
good point, done.

Reply:
good point, done